File: question/leetcode/leetcode0070.Climbing-Stairs/leetcode0070.go
package leetcode

func climbStairs(n int) int {
	if n <= 2 {
		return n
	}

	prePre, pre := 1, 2
	var curr int
	for i := 3; i <= n; i++ {
		curr = prePre + pre
		prePre, pre = pre, curr
	}

	return curr
}

File: question/leetcode/leetcode0070.Climbing-Stairs/leetcode0070.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 2:
            return n
        pre_pre, pre = 1, 2
        cur = 0
        for i in range(3, n+1):
            cur = pre_pre + pre
            pre_pre, pre = pre, cur

        return cur

